js/map_utils.js

```diff
@@ -258,3 +260,67 @@ function updateCircles(activeCategory) {
       .attr("r", function(d) { return scaleCircles(d.properties[[activeCategory]]); });
 }
 
+function seeTooltip(currentCircle, d) {
+  var orig = d3.select(currentCircle),
+      origNode = orig.node();
+  var duplicate = d3.select(origNode.parentNode.appendChild(origNode.cloneNode(true), 
+                                                            origNode.nextSibling));
+  
+  // style circles
+  orig
+    .style("opacity", 0); // makes original circle invisible in the background
+  duplicate
+    .classed('county-point-duplicate', true)
+    .style("pointer-events", "none")
+    .style("opacity", 1); // makes the duplicate circle on the top
+  
+  // change tooltip
+  d3.select(".tooltip")
+    .classed("shown", true)
+    .classed("hidden", false)
+    .transition()
+    .duration(50)
+    .style("left", (d3.event.pageX + 35) + "px")
+    .style("top", (d3.event.pageY - 50) + "px");
+  d3.select(".tooltip")
+    .html(d.properties.COUNTY + "<br/>" + 
+            "Population: " + d.properties.countypop + "<br/>" +
+            categoryToName(activeCategory) + ": " + 
+              d.properties[[activeCategory]] + " " + "MGD");
+}
+
+function hideTooltip(currentCircle, d) {
+  d3.select(currentCircle)
+    .style("opacity", 0.8);
+  d3.select('.county-point-duplicate')
+    .remove(); // delete duplicate
+  d3.select(".tooltip")
+    .classed("shown", false)
+    .classed("hidden", true)
+    .transition()
+    .duration(100);
+}
+
+d3.selection.prototype.moveToFront = function() {  
+      return this.each(function(){
+        this.parentNode.appendChild(this);
+      });
+    };
+
+function categoryToClass(category) {
```

Comment:
this is getting to be a lot of ways of expressing category. can we keep it to one internal representation (classes, hash tags, etc.) and one we show the world?

Reply:
^ yes I would like that. @mwernimont can we update the classes to be what is in the data already:
```
total
thermoelectric
publicsupply
irrigation
industrial
```

Reply:
For the buttons?

Reply:
Marty and I talked. Yes, we would like to try to get the class to be the same, but realized that we need probably need multiple class definitions anyways since buttons need the color defined by `background` and the circles need `fill`

Reply:
makes sense. so what's your plan?